pyannote/audio/interactive/embeddings/recipe.py

```diff
@@ -0,0 +1,426 @@
+    source: Path,
+    labels: List[str],
+    chunk: float = 30.0,
+    randomize: bool = False,
+) -> Iterable[Dict]:
+    """Stream for `pyannote.audio` recipe
+
+    Applies pretrained pipeline and sends the results for manual correction
+
+    Parameters
+    ----------
+    pipeline : Pipeline
+        Pretrained pipeline.
+    source : Path
+        Directory containing audio files to process.
+    labels : list of string
+        List of expected pipeline labels.
+    chunk : float, optional
+        Duration of chunks, in seconds. Defaults to 30s.
+
+    Yields
+    ------
+    task : dict
+        Prodigy task with the following keys:
+        "path" : path to audio file
+        "chunk" : chunk start and end times
+        "audio" : base64 encoding of audio chunk
+        "text" : chunk identifier "{filename} [{start} {end}]"
+        "audio_spans" : list of audio spans {"start": ..., "end": ..., "label": ...}
+        "audio_spans_original" : deep copy of "audio_spans"
+        "meta" : metadata displayed in Prodigy UI {"file": ..., "start": ..., "end": ...}
+        "config": {"labels": list of labels}
+    """
+
+    context = getattr(pipeline, "context", 2.5)
+
+    audio_for_prodigy = AudioForProdigy()
+    audio_for_pipeline = Audio(mono=True)
+
+    chunks = get_chunks(source, chunk_duration=chunk)
+    if randomize:
+        chunks = list(chunks)
+        random.shuffle(chunks)
+
+    for file, excerpt in chunks:
+
+        path = file["path"]
+        filename = file["text"]
+        text = f"{filename} [{excerpt.start:.1f} - {excerpt.end:.1f}]"
+        print("--------------------------New FILE")
+
+        # load contextualized audio excerpt
+        excerpt_with_context = Segment(
+            start=excerpt.start - context, end=excerpt.end + context
+        )
+        waveform_with_context, sample_rate = audio_for_pipeline.crop(
+            path, excerpt_with_context, mode="pad"
+        )
+
+        # run pipeline on contextualized audio excerpt
+        output: Annotation = pipeline(
+            {"waveform": waveform_with_context, "sample_rate": sample_rate}
+        )
+
+        # crop, shift, and format output for visualization in Prodigy
+        audio_spans = get_audio_spans(
+            output, excerpt, excerpt_with_context=excerpt_with_context
+        )
+
+        # load audio excerpt for visualization in Prodigy
+        audio = audio_for_prodigy.crop(path, excerpt)
+
+        alllabels = sorted(set(labels) | set(output.labels()))
+
+        # group by label
+        print("Speaker recorded : ", nSpeakerVoices["name"])
+        audio_spans = sorted(audio_spans, key=lambda x: x["label"])
+        for label, segments in groupby(audio_spans, key=lambda x: x["label"]):
+
+            combine_waveform = torch.Tensor([])
+            for segment in list(segments):
+                waveform, sample_rate = audio_for_pipeline.crop(
+                    path, Segment(segment["start"], segment["end"])
```

Comment:
`segment["start"]` and `segment["end"]` are therefore also between `0.0` and `chunk`, right?

However, `path` is the whole file... not the current chunk.
So you will always read the beginning of the file...